### PR TITLE
Bookstore: Use Redux in React components.

### PR DESCRIPTION
**Pull request by JuanDBta**
In this branch, I used a Redux store to display books. Also, I implemented react-redux in order to connect the store and then access state variables (accessed through useSelector) and state modifiers (which will be a combination of a reducer and useDispatch).

Additionally, you can find the following features:

Wrapped App with the Provider component from react-redux
Passed configured store into the Provider
Create an initialState variable for booksSlice.js,, which includes a books object.

Display books, received from the slice, in a reusable component
Dispatch actions using useDispatch
Add a <Button> component, which includes:
--An event handler that adds a book to the books array (with attributes id, title, and author)

Add a <Button> component, which includes:
--An event handler that removes a book from the books array (by id)

In order to work with Redux successfully I deleted some React components that became unnecessary.

### CODE REVIEW

File: src/redux/features/books/booksSlice.js [hunk 1]
```diff
@@ -1,18 +1,44 @@
...
   reducers: {
     addBook: (state, action) => {
-      state.books.push(action.payload);
+      const newBook = {
+        itemId: uuidv4(),
+        title: action.payload.title,
+        author: action.payload.author,
+        category: action.payload.category,
+      };
+      state.books.push(newBook);
```
Comment (Olamarx) [inline src/redux/features/books/booksSlice.js]: - [ ] The addBook function is not working and I suspect it is from here, I will encourage you to change this to `state.books = [...state.books, action.payload]` and by that, we are not mutating the state and we are taking advantage of the spread operator to add to the books present there before. As it stands, I cant add to the books. Once I try to, it loads the initial state data only.

File: src/components/Books.jsx [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+  const [category, setCategory] = useState('');
+
+  const handleAddBook = () => {
+    const newBook = {
+      itemId: uuidv4(),
+      title,
+      author,
+      category,
+    };
+    dispatch(addBook(newBook));
```
Comment (Olamarx) [inline src/components/Books.jsx:20]: This is cool.

File: src/App.js [hunk 3]
```diff
@@ -1,35 +1,27 @@
...
-import Books from './redux/features/books/Books';
-import Categories from './redux/features/categories/Categories';
 
 function App() {
-/*  const router = createBrowserRouter(
+  const router = createBrowserRouter(
     createRoutesFromElements(
       <Route path="/" element={<Root />}>
         <Route path="/bookstore" element={<BookStore />} />
         <Route path="/categories" element={<Categories />} />
```
Comment (Olamarx) [inline src/App.js]: - [OPTIONAL] I will advise you to use the BookStore as the root route so that it can be user friendly and understanding.

**Review by Olamarx — CHANGES_REQUESTED** [review]
Hi @JuanDBta, 

Good job so far, indeed you have spent time on this project!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
- Linters passed ✔️
- Readme file professional ✔️
- Remove function ✔️
- All methods in place ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by nedjwab — APPROVED** [review]
# 🟢 Project Approved! 🏆

## Hi  @JuanDBta      👋

Your project is complete! ✔️✔ , I loved your work, it looks great 🔥🔥🔥🔥 🏁

Your code was very professional 💯

There is nothing else to say other than... _it's time to merge it_ :shipit:
Congratulations! 🎉

 
## Highlights 🌟

- ✔️ Descriptive commmit messsages
- ✔️ Descriptive PR description
- ✔️ No linter Errors

- The code is top-notch! 💯 and have a huge potential.

Cheers and Happy coding!👏👏👏

## Important Notes 📝

- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.


_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by JuanDBta** [comment]
Thank you for the lovely comments!

**Pull request merged**